src/status_im/ui/screens/wallet/subs.cljs

```diff
@@ -14,22 +14,39 @@
     (:balance wallet)))
 
 (re-frame/reg-sub :price
-  (fn [db]
-    (get-in db [:prices :price])))
+  (fn [db [_ fsym tsym]]
+    (get-in db [:prices fsym tsym :price])))
 
 (re-frame/reg-sub :last-day
-  (fn [db]
-    (get-in db [:prices :last-day])))
+  (fn [db [_ fsym tsym]]
+    (get-in db [:prices fsym tsym :last-day])))
+
+(re-frame/reg-sub :asset-value
+  (fn [[_ fsym tsym]]
+    [(re-frame/subscribe [:balance])
+     (re-frame/subscribe [:price fsym tsym])])
+  (fn [[balance price] [_ fsym tsym]]
+    (when (and balance price)
+      (-> (money/wei->ether (get balance fsym))
+          (money/symbol->fiat price)
+          (money/with-precision 2)
+          str))))
 
 (re-frame/reg-sub :portfolio-value
   :<- [:balance]
-  :<- [:price]
-  (fn [[balance price]]
-    (if (and balance price)
-      (-> (money/wei->ether (get balance :ETH)) ;; TODO(jeluard) Modify to consider tokens
-          (money/eth->usd price)
-          (money/with-precision 2)
-          str)
+  :<- [:get :prices]
```

Comment:
We tend no to use `[:get :something]` type of subscriptions, and prefer custom ones like `[:prices]` for example. You can then reuse that in `:price`.